### PR TITLE
Add cleanup script to base images 

### PR DESCRIPTION
**Pull request by romoh**
Remove /srv linking that is created by the toolchain & filesystems package.  Toolkit and filesystem package unnecessarily symlink /var/srv & /srv which interferes with other scenarios. This will be fixed in Mariner 3.0, this PR is a mitigation for 2.0 images.

More details:
> ln -svfn var/srv %{buildroot}/srv
srv symlink created by the toolkit https://github.com/microsoft/CBL-Mariner/blob/4e4223205fdd0154713821430f2224290ff94a74/toolkit/scripts/toolchain/container/toolchain_initial_chroot_setup.sh#L41
The symlink is also created by  https://github.com/microsoft/CBL-Mariner/blob/4e4223205fdd0154713821430f2224290ff94a74/SPECS/filesystem/filesystem.spec#L63 

On any mariner image, the symlink is left behind which is interfering with scenarios where customers want to mount a volume over /srv

PR also adds cleanup for any logs that may be left after the build process is completed.

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?

###### Does this affect the toolchain?  

**NO**

###### Test Methodology
Built locally and validated.

### CODE REVIEW

File: toolkit/imageconfigs/scripts/cleanup.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+else
+  echo "/srv symlink does not exist"
+fi
+
+# cleanup any logs that may have been created during the build
+if [ -L /var/log ]; then
+  echo "Clearing /var/log"
+  rm -rf /var/log/*
+else
+  echo "/var/log does not exist"
```
Comment (vinceaperri) [inline toolkit/imageconfigs/scripts/cleanup.sh:18]: /var/log symlink*

**Review by vinceaperri — APPROVED** [review]

File: toolkit/imageconfigs/scripts/cleanup.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+if [ -L /srv ]; then
+  echo "Removing /srv symlink"
+  rm /srv
+else
+  echo "/srv symlink does not exist"
+fi
+
+# cleanup any logs that may have been created during the build
+if [ -L /var/log ]; then
+  echo "Clearing /var/log"
```
Comment (vinceaperri) [inline toolkit/imageconfigs/scripts/cleanup.sh:15]: Should this be "Removing /var/log/ symlink" too?

File: toolkit/imageconfigs/scripts/cleanup.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+# cleanup symlinks created by the toolkit that are not needed for base images
+if [ -L /srv ]; then
+  echo "Removing /srv symlink"
+  rm /srv
+else
+  echo "/srv symlink does not exist"
+fi
+
+# cleanup any logs that may have been created during the build
```
Comment (vinceaperri) [inline toolkit/imageconfigs/scripts/cleanup.sh:13]: With this comment, it seems the intention is to just "rm -rf /var/log" if it exists / is a symlink.

**Review by vinceaperri — APPROVED** [review]

File: toolkit/imageconfigs/scripts/cleanup.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
```
Comment (neha170) [inline toolkit/imageconfigs/scripts/cleanup.sh:2]: Please add the license banner
`
# Copyright (c) Microsoft Corporation.
# Licensed under the MIT License.
`

File: toolkit/imageconfigs/scripts/cleanup.sh (hunk 3 above)
Comment (romoh) [reply toolkit/imageconfigs/scripts/cleanup.sh:13]: Here I'm actually clearing the logs directory and not removing symlinks

File: toolkit/imageconfigs/scripts/cleanup.sh (hunk 4 above)
Comment (romoh) [reply toolkit/imageconfigs/scripts/cleanup.sh:2]: good reminder. 

I see other scripts under `postinstallscripts` don't have it though. so wondering what's the criteria for that?

File: toolkit/imageconfigs/scripts/cleanup.sh (hunk 2 above)
Comment (romoh) [reply toolkit/imageconfigs/scripts/cleanup.sh:18]: see my other comment

File: toolkit/imageconfigs/scripts/cleanup.sh (hunk 1 above)
Comment (romoh) [reply toolkit/imageconfigs/scripts/cleanup.sh:21]: see my other comment. This is not removing symlinks

File: toolkit/imageconfigs/scripts/cleanup.sh (hunk 3 above)
Comment (romoh) [reply toolkit/imageconfigs/scripts/cleanup.sh:16]: ah actually I just noticed I got the check wrong. Will fix.

File: toolkit/imageconfigs/scripts/cleanup.sh (hunk 4 above)
Comment (neha170) [reply toolkit/imageconfigs/scripts/cleanup.sh:2]: AFAIK, all scripts should be carrying it

**Comment by cwize1** [comment]
The '/srv' -> '/var/srv' symlink is created by the `filesystem` package, not the toolchain.

File: toolkit/imageconfigs/scripts/cleanup.sh [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+set -eux
+
+# cleanup symlinks created by the toolkit that are not needed for base images
```
Comment (cwize1) [inline toolkit/imageconfigs/scripts/cleanup.sh:8]: You should add a code comment saying why this is needed. Considering that this symlink is being added by the `filesystem` package, this seems like intentional behavior for Mariner. So, there should be justification for why the bare-metal and qemu-guest images need to be different.

Also, add this comment to the PR description as well.

File: toolkit/imageconfigs/scripts/cleanup.sh [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  echo "Removing /srv symlink"
+  rm /srv
+else
+  echo "/srv symlink does not exist"
+fi
+
+# cleanup any logs that may have been created during the build
+if [ -d /var/log ]; then
+  echo "Clearing /var/log"
+  rm -rf /var/log/*
```
Comment (cwize1) [inline toolkit/imageconfigs/scripts/cleanup.sh:19]: Out of interest, what files are getting added to `/var/log` during the build?

**Review by cwize1 — CHANGES_REQUESTED** [review]

File: toolkit/imageconfigs/scripts/cleanup.sh (hunk 5 above)
Comment (romoh) [reply toolkit/imageconfigs/scripts/cleanup.sh:8]: updated from a logged bug.

File: toolkit/imageconfigs/scripts/cleanup.sh (hunk 6 above)
Comment (romoh) [reply toolkit/imageconfigs/scripts/cleanup.sh:19]: Nothing specific that I'm aware of. Trying to align to https://github.com/microsoft/CBL-Mariner/blob/main/toolkit/imageconfigs/postinstallscripts/core-container/cleanup.sh where it makes sense.

**Comment by romoh** [comment]
> The '/srv' -> '/var/srv' symlink is created by the `filesystem` package, not the toolchain.

It is created by both.

**Review by cwize1 — APPROVED** [review]

**Review by vinceaperri — APPROVED** [review]

**Pull request merged**